packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-payment-strategy.ts

```diff
@@ -0,0 +1,112 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import {
+    ContentType,
+    INTERNAL_USE_ONLY,
+    InvalidArgumentError,
+    OrderRequestBody,
+    PaymentArgumentInvalidError,
+    PaymentIntegrationService,
+    SDK_VERSION_HEADERS,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import GooglePayPaymentProcessor from '../google-pay-payment-processor';
+import GooglePayPaymentStrategy from '../google-pay-payment-strategy';
+import { GooglePayInitializationData } from '../types';
+
+import PayPalCommerceScriptLoader from './google-pay-paypal-commerce-script-loader';
+import { ConfirmOrderData } from './types';
+
+export default class GooglePayPaypalCommercePaymentStrategy extends GooglePayPaymentStrategy {
+    constructor(
+        _paymentIntegrationService: PaymentIntegrationService,
+        _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+        private _paypalCommerceScriptLoader: PayPalCommerceScriptLoader,
+        private _requestSender: RequestSender,
+    ) {
+        super(_paymentIntegrationService, _googlePayPaymentProcessor);
+    }
+
+    async execute({ payment }: OrderRequestBody): Promise<void> {
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const cartId = this._paymentIntegrationService.getState().getCartOrThrow().id;
+
+        const { orderId } = await this.getOrderId(cartId);
+        const { initializationData } = this._paymentIntegrationService
+            .getState()
+            .getPaymentMethodOrThrow<GooglePayInitializationData>(this._getMethodId());
```

Comment:
To avoid code duplication this code can be simplified:
```suggestion
        const state = this._paymentIntegrationService.getState()
        const cartId = state.getCartOrThrow().id;
        const { initializationData } = state.getPaymentMethodOrThrow<GooglePayInitializationData>(this._getMethodId());
        const { orderId } = await this.getOrderId(cartId);
```